src/pfctl_worker.cpp

```diff
@@ -20,206 +20,206 @@ using namespace std;
-				"pfctl"
-			);
-			rmq = &mq;
-		}
-		catch(const interprocess_exception& ex)
-		{
-			if (ex.get_error_code() == 7) {
-				// Awaiting master process to create the queue
-				usleep(100000);
-				continue;
-			} else {
-				log(MSG_INFO, fmt::sprintf("pfctl_worker: Interprocess Exception while waiting for queue %d", ex.get_error_code()));
-			}
-		}
-		catch(const runtime_error& ex)
-		{
-			// speciffic handling for runtime_error
-			log(MSG_INFO, fmt::sprintf("pfctl_worker: Exception while waiting for queue %s", ex.what()));
-		}
-		catch(const exception& ex)
-		{
-			// std::runtime_error which is handled explicitly
-			log(MSG_INFO, fmt::sprintf("pfctl_worker: Exception while waiting for queue %s", ex.what()));
-		}
-		break;
-	}
-
-	return rmq;
+message_queue *attach_pfctl_queue() {
+  message_queue *rmq = NULL;
+  while (true) {
+    try {
+      static message_queue mq(open_only, "pfctl");
+      rmq = &mq;
+    } catch (const interprocess_exception &ex) {
+      if (ex.get_error_code() == 7) {
+        // Awaiting master process to create the queue
+        usleep(100000);
+        continue;
+      } else {
+        log(MSG_INFO, fmt::sprintf("pfctl_worker: Interprocess Exception while "
+                                   "waiting for queue %d",
+                                   ex.get_error_code()));
+      }
+    } catch (const runtime_error &ex) {
+      // speciffic handling for runtime_error
+      log(MSG_INFO,
+          fmt::sprintf("pfctl_worker: Exception while waiting for queue %s",
+                       ex.what()));
+    } catch (const exception &ex) {
+      // std::runtime_error which is handled explicitly
+      log(MSG_INFO,
+          fmt::sprintf("pfctl_worker: Exception while waiting for queue %s",
+                       ex.what()));
+    }
+    break;
+  }
+
+  return rmq;
 }
 
-bool send_message(message_queue* mq, string pool_name, string table_name, set<LbNode*> lb_nodes) {
-	pfctl_msg msg;
-
-	memset(&msg, 0, sizeof(msg));
-	strncpy(msg.pool_name, pool_name.c_str(), sizeof(msg.pool_name));
-	strncpy(msg.table_name, table_name.c_str(), sizeof(msg.table_name));
-	int node_index = 0;
-	assert(lb_nodes.size() < MAX_NODES);
-	for (auto node: lb_nodes) {
-		strncpy(msg.wanted_addresses[node_index], node->address.c_str(), ADDR_LEN);
-		node_index++;
-	}
-	return mq->try_send(&msg, sizeof(pfctl_msg), 0);
+bool send_message(message_queue *mq, string pool_name, string table_name,
+                  set<LbNode *> lb_nodes) {
+  pfctl_msg msg;
+
+  memset(&msg, 0, sizeof(msg));
+  strncpy(msg.pool_name, pool_name.c_str(), sizeof(msg.pool_name));
+  strncpy(msg.table_name, table_name.c_str(), sizeof(msg.table_name));
+  int node_index = 0;
+  assert(lb_nodes.size() < MAX_NODES);
+  for (auto node : lb_nodes) {
+    if (!node->ipv4_address.empty()) {
+      strncpy(msg.wanted_addresses[node_index], node->ipv4_address.c_str(),
+              ADDR_LEN);
+      node_index++;
+    }
+    if (!node->ipv4_address.empty()) {
```

Comment:
wrong if! this is a bug. it should be ipv6 here

Reply:
Fixed.